hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterHeartbeatService.java

```diff
@@ -76,6 +75,8 @@ public void setup() throws Exception {
     curatorFramework.start();
     routerConfig.set(CommonConfigurationKeys.ZK_ADDRESS, connectStr);
     router.init(routerConfig);
+    // set custom routerid after router.init()
+    router.setRouterId(routerId);
```

Comment:
Wouldnt init already trigger this with the new code change?
Or we need to force the specific routerId?

Reply:
If set the specific routerid,we use the specific routerid; otherwise we use the routerid generated by Router.serviceInit.